kyaml/kio/kio.go

```diff
@@ -111,7 +117,7 @@ func (p Pipeline) ExecuteWithCallback(callback PipelineExecuteCallbackFunc) erro
 		// TODO (issue 2872): This len(result) == 0 should be removed and empty result list should be
 		// handled by outputs. However currently some writer like LocalPackageReadWriter
 		// will clear the output directory and which will cause unpredictable results
-		if len(result) == 0 || err != nil {
+		if len(result) == 0 && !p.ContinueOnEmptyResult || err != nil {
```

Comment:
As mentioned in the TODO comment, pipeline shouldn't stop when the result is empty in intermediate steps. This `len(result) == 0` is left here because the behavior in `LocalPackageReadWriter`. Please update the TODO comment. And also please add a note to the comment for `ContinueOnEmptyResult` to notice that this option is a temporary work around and may be removed in future.

Reply:
A `Pipeline` should be somewhat configurable; this seems like a reasonable knob to have.

The comment for `ContinueOnEmptyResult` might be improved with a use case for continuing;
presumably a to allow a generator later in the pipe to do its job.

`Pipeline` configuration is a high level API, so we should try to settle how to handle this before 1.0.

Reply:
Pipeline is often used without LocalPackageReadWriter, so this option should be available.  LocalPackageReadWriter is an implementation of an interface and shouldn't be tightly coupled to Pipeline.